backend/src/sms/routes/tests/sms-campaign.routes.test.ts

```diff
@@ -0,0 +1,238 @@
+        message:
+          'Please re-upload your recipient list as template has changed.',
+        extra_keys: ['variable2'],
+        num_recipients: 0,
+        template: {
+          body: 'test {{variable2}}',
+          params: ['variable2'],
+        },
+      })
+    )
+  })
+
+  test('Fail to update template for SMS campaign', async () => {
+    const campaign = await Campaign.create({
+      name: 'campaign-1',
+      userId: 1,
+      type: 'SMS',
+      valid: false,
+      protect: false,
+    })
+
+    const res = await request(app)
+      .put(`/campaign/${campaign.id}/sms/template`)
+      .query({ campaignId: campaign.id })
+      .send({
+        body: '<p></p>',
+      })
+    expect(res.status).toBe(400)
+    expect(res.body).toEqual({
+      message:
+        'Message template is invalid as it only contains invalid HTML tags!',
+    })
+  })
+})
+
+describe('GET /campaign/{id}/sms/upload/start', () => {
+  test('Fail to generate presigned URL when invalid md5 provided', async () => {
+    const campaign = await Campaign.create({
+      name: 'campaign-1',
+      userId: 1,
+      type: 'SMS',
+      valid: false,
+      protect: false,
+    })
+
+    UploadService.getUploadParameters = jest
+      .fn()
+      .mockRejectedValue({ message: 'hello' })
+
+    const res = await request(app)
+      .get(`/campaign/${campaign.id}/sms/upload/start`)
+      .query({
+        mime_type: 'text/csv',
+        md5: 'invalid md5 checksum',
+      })
+
+    expect(res.status).toBe(500)
+    expect(res.body).toEqual({
+      message: 'Unable to generate presigned URL',
+    })
+  })
+
+  test('Successfully generate presigned URL for valid md5', async () => {
+    const campaign = await Campaign.create({
+      name: 'campaign-1',
+      userId: 1,
+      type: 'SMS',
+      valid: false,
+      protect: false,
+    })
+
+    UploadService.getUploadParameters = jest
+      .fn()
+      .mockReturnValue({ presignedUrl: 'url', signedKey: 'key' })
+
+    const res = await request(app)
+      .get(`/campaign/${campaign.id}/sms/upload/start`)
+      .query({
+        mime_type: 'text/csv',
+        md5: 'valid md5 checksum',
+      })
+
+    expect(res.status).toBe(200)
+    expect(res.body).toEqual({ presigned_url: 'url', transaction_id: 'key' })
+  })
+})
+
+describe('POST /campaign/{id}/sms/upload/complete', () => {
+  test('Fails to complete upload if template is missing', async () => {
+    const campaign = await Campaign.create({
+      name: 'campaign-1',
+      userId: 1,
+      type: 'SMS',
+      valid: false,
+      protect: false,
+    })
+
+    UploadService.extractParamsFromJwt = jest
+      .fn()
+      .mockReturnValue({ s3Key: 'key' })
+
+    const res = await request(app)
+      .post(`/campaign/${campaign.id}/sms/upload/complete`)
+      .send({ transaction_id: '123', filename: 'abc', etag: '123' })
+
+    expect(res.status).toEqual(500)
+  })
+
+  test('Successfully starts to complete upload', async () => {
```

Comment:
```suggestion
  test('Successfully starts recipient list processing', async () => {
```

Minor nit. This might be clearer.